src/scan_nm2.cc

```diff
@@ -0,0 +1,353 @@
+
+            if (auto ret = process.run()) {
+                if (auto res = process.wait()) {
+                    if (*res == 0) {
+                        credId = cred->getId();
+                        output = process.readAllStandardOutput();
+                        break;
+                    }
+                } else {
+                    logWarn("Cannot wait 'etn-nut-powerconnect', error: {}", res.error());
+                }
+            } else {
+                logWarn("Cannot run 'etn-nut-powerconnect', error: {}", ret.error());
+            }
+        }
+
+        if (!output.empty()) {
+            std::map<std::string, std::string> dump;
+            std::stringstream                  ss(output);
+            for (std::string line; std::getline(ss, line);) {
+                auto [key, value] = fty::split<std::string, std::string>(line, rex);
+                dump.emplace(key, value);
+            }
+
+            nut_dumpdata_to_fty_message(msgs, dump, m_mapping, m_sensorMapping, m_address, "device");
+            logDebug("Dump data for {} ('nut_powercom') succeeded.", m_address);
+
+            for (const auto& asset : msgs) {
+                // get asset subtype
+                std::string subtype(fty_proto_aux_string(asset, "subtype", ""));
+                std::string type(fty_proto_aux_string(asset, "type", ""));
+                std::string sensor;
+                if (subtype == "sensor") {
+                    // if the device is a sensor, but the model is not present, we need to fill the field anyway
+                    sensor = fty_proto_ext_string(asset, "model", "unknown");
+                }
+
+                logDebug("Processing asset {} ({} - {})", fty_proto_aux_string(asset, "name", "iname"), type, subtype);
+
+                if (sensor.empty()) { // not a sensor
+                    std::string daisyChain(fty_proto_ext_string(asset, "daisy_chain", ""));
+                    fty_proto_ext_insert(asset, "endpoint.1.sub_address", (daisyChain == "0") ? "" : daisyChain.c_str());
+                    fty_proto_ext_insert(asset, "endpoint.1.protocol", "nut_powercom");
+                    fty_proto_ext_insert(asset, "endpoint.1.nut_powercom.secw_credential_id", credId.c_str());
+                } else {
+                    std::string modbusAddress(fty_proto_ext_string(asset, "modbus_address", ""));
+                    if (sensor == "EMPDT1H1C2") {
+                        fty_proto_ext_insert(asset, "endpoint.1.sub_address", modbusAddress.c_str());
+                    } else {
+                        logWarn("Sensor model {} is not supported", sensor);
+                    }
+                }
+            }
+        }
+
+        return !output.empty();
+    }
+
+    fty::Expected<std::string> findExecutable(const std::string& name) const
+    {
+        static std::vector<std::filesystem::path> paths =
+            {"/usr/lib/nut", "/lib/nut", "/home/jes/workspace/fty/build/Debug/deps-runtime/bin"};
```

Comment:
would be worth a conf file or cmd line param to be able to add developer's own path like yours @etnKlendathu  ;)